smartsim/settingshold/translators/batch/lsf.py

```diff
@@ -0,0 +1,145 @@
+        """
+        # For compatibility with other launchers, as explained in docstring
+        if walltime:
+            if len(walltime.split(":")) > 2:
+                walltime = ":".join(walltime.split(":")[:2])
+        return {"W": walltime}
+
+    def set_smts(self, smts: int) -> t.Union[IntegerArgument,None]:
+        """Set SMTs
+
+        This sets ``-alloc_flags``. If the user sets
+        SMT explicitly through ``-alloc_flags``, then that
+        takes precedence.
+
+        :param smts: SMT (e.g on Summit: 1, 2, or 4)
+        """
+        return {"alloc_flags": int(smts)}
+
+    def set_project(self, project: str) -> t.Union[StringArgument,None]:
+        """Set the project
+
+        This sets ``-P``.
+
+        :param time: project name
+        """
+        return {"P": project}
+    
+    def set_account(self, account: str) -> t.Union[StringArgument,None]:
+        """Set the project
+
+        this function is an alias for `set_project`.
+
+        :param account: project name
+        """
+        return self.set_project(account)
+
+    def set_nodes(self, num_nodes: int) -> t.Union[IntegerArgument,None]:
+        """Set the number of nodes for this batch job
+
+        This sets ``-nnodes``.
+
+        :param nodes: number of nodes
+        """
+        return {"nnodes": int(num_nodes)}
+
+    def set_hostlist(self, host_list: t.Union[str, t.List[str]]) -> t.Union[StringArgument,None]:
+        """Specify the hostlist for this job
+
+        :param host_list: hosts to launch on
+        :raises TypeError: if not str or list of str
+        """
+        if isinstance(host_list, str):
+            host_list = [host_list.strip()]
+        if not isinstance(host_list, list):
+            raise TypeError("host_list argument must be a list of strings")
+        if not all(isinstance(host, str) for host in host_list):
+            raise TypeError("host_list argument must be list of strings")
+        return {"m": '"' + " ".join(host_list) + '"'}
+
+    def set_tasks(self, tasks: int) -> t.Union[IntegerArgument,None]:
+        """Set the number of tasks for this job
+
+        This sets ``-n``
+
+        :param tasks: number of tasks
+        """
+        return {"n": int(tasks)}
+
+    def set_queue(self, queue: str) -> t.Union[StringArgument,None]:
+        """Set the queue for this job
+        
+        This sets ``-q``
+
+        :param queue: The queue to submit the job on
+        """
+        return {"q": queue}
+
+    def format_batch_args(self, batch_args: t.Dict[str, t.Union[str,int,float,None]]) -> t.List[str]:
+        """Get the formatted batch arguments for a preview
+
+        :return: list of batch arguments for Qsub
+        """
+        opts = []
+
+        for opt, value in batch_args.items():
+
+            prefix = "-"  # LSF only uses single dashses
+
+            if not value:
+                opts += [prefix + opt]
+            else:
+                opts += [" ".join((prefix + opt, str(value)))]
```

Comment:
Thoughts on changing this to `opts += [f"{prefix}{opt}", str(value)]` instead? I'd rather have each symbol be its own entry in the list so that if someone where to naively `Popen` the command it behaves as expected.

Other wise we could run into weird divergent behavior similar to `echo -n hi`  vs `echo '-n hi'`

Reply:
should this be handled the same for the slurm class, where it is `opts += ["=".join((prefix + opt, str(value)))]`? The conversion will be handled later on?

Reply:
Its less important there because `'--some-arg=some-value'` (note the `'`s) will be treated equivalent to `--some-arg=some-value` by `Popen`/a shell (assuming that is where these options are eventually going to get sent).

`'--some-arg some-val'` (again note the `'`s) though can treated as very to `--some-arg some-val`. The former is 1 symbol while the later is 2 (`'--some-arg'` and `'some-val'`).

The general rule of thumb when creating lists of args that may be sent to a subprocess is have each symbol as its own string (see [`man execv`](https://linux.die.net/man/3/execv) family of functions for more detail, which iirc is what python calls under the hood), but we got kinda cavalier about this in the original SS impl and I kinda want to fix it while we have the chance!

Reply:
That said, I think
```py
f"{prefix}{opt}={value}"
```
is WAYYYY easier to read than
```py
" ".join((prefix + opt, str(value)))
```
so if you're willing to make the change I say go for it!